cmd/go-yaml/cli_test.go

```diff
@@ -0,0 +1,193 @@
+package main
+
+import (
+	"bytes"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"strings"
+	"testing"
+
+	"go.yaml.in/yaml/v4"
+)
+
+// TestCase represents a single test case from a test file
+type TestCase struct {
+	Name  string `yaml:"name"`
+	Text  string `yaml:"text"`
+	Token string `yaml:"token,omitempty"`
+	TOKEN string `yaml:"TOKEN,omitempty"`
+	Event string `yaml:"event,omitempty"`
+	EVENT string `yaml:"EVENT,omitempty"`
+	Node  string `yaml:"node,omitempty"`
+	NODE  string `yaml:"NODE,omitempty"`
+	Yaml  string `yaml:"yaml,omitempty"`
+	YAML  string `yaml:"YAML,omitempty"`
+	Json  string `yaml:"json,omitempty"`
+	JSON  string `yaml:"JSON,omitempty"`
+}
+
+// TestSuite is a sequence of test cases
+type TestSuite []TestCase
+
+// flagMapping maps test file field names to CLI flags
+var flagMapping = map[string]string{
+	"token": "-t",
+	"TOKEN": "-T",
+	"event": "-e",
+	"EVENT": "-E",
+	"node":  "-n",
+	"NODE":  "-N",
+	"yaml":  "-y",
+	"YAML":  "-Y",
+	"json":  "-j",
+	"JSON":  "-J",
+}
+
+func TestCLI(t *testing.T) {
+	// Find all test files in testdata/
+	testFiles, err := filepath.Glob("testdata/*.yaml")
+	if err != nil {
+		t.Fatalf("Failed to find test files: %v", err)
+	}
+
+	if len(testFiles) == 0 {
+		t.Skip("No test files found in testdata/")
+	}
+
+	// Build the CLI binary if it doesn't exist
+	binaryPath := "../../go-yaml"
+	if _, err := os.Stat(binaryPath); os.IsNotExist(err) {
```

Comment:
You should stop here and report an error if the file exists but is a folder.

Think about someone who created a go-yaml folder.